Assignment4/sparql-T110304.txt

```diff
@@ -0,0 +1,79 @@
+- Get all the classes:
+
+select distinct ?class where{?class a <http://www.w3.org/2000/01/rdf-schema#Class>}
+
+- Get all the subclasses of the class Establishment:
+
+select distinct ?subClass where {?subClass <http://www.w3.org/2000/01/rdf-schema#subClassOf> <http://GP-onto.fi.upm.es/exercise2#Establishment>}
+
+- Get all instances of the class City
+
+select distinct ?ListInstances where{ ?ListInstances a <http://GP-onto.fi.upm.es/exercise2#City>}
+
+- Get the number of inhabitants of Santiago de Compostela
+
+select distinct  ?number where{<http://GP-onto.fi.upm.es/exercise2#Santiago_de_Compostela> <http://GP-onto.fi.upm.es/exercise2#hasInhabitantNumber> ?number}
+
+- Get the number of inhabitants of Santiago de Compostela and Arzua
+
+
+select distinct ?SantiagoDeCompostela,?Arzua where{<http://GP-onto.fi.upm.es/exercise2#Santiago_de_Compostela> <http://GP-onto.fi.upm.es/exercise2#hasInhabitantNumber> ?SantiagoDeCompostela. <http://GP-onto.fi.upm.es/exercise2#Arzua> <http://GP-onto.fi.upm.es/exercise2#hasInhabitantNumber> ?Arzua .}
+
+
+- Get all places, together with the number of inhabitants, ordered by the place name (ascending)
+
+select distinct ?place,?number where{ ?placeOrd <http://GP-onto.fi.upm.es/exercise2#hasInhabitantNumber> ?number;
+<http://www.w3.org/2000/01/rdf-schema#label> ?place}
+order by asc(?placeOrd)
+
+
+- Get all instances of  Locality together with their number of inhabitants (if this information exists)
+
+
+select distinct ?local,?number where{ ?localOr <http://GP-onto.fi.upm.es/exercise2#Locality> ?number;
+<http://www.w3.org/2000/01/rdf-schema#label> ?local}
+optional {?number}
+
+
+- Get all places with more than 200.000 inhabitants
+
+
+select distinct ?number,?places where{ ?places <http://GP-onto.fi.upm.es/exercise2#hasInhabitantNumber > ?number
+filter (?number > 200000)}
+

```

Comment:
Did this query work? I don't think that it can be even parsed. 
The same applies to most of the following queries.
Please check them first.